ttyplot.c

```diff
@@ -333,16 +325,18 @@ static void signal_handler(int signum) {
 }
 
 static void redraw_screen(const char *errstr) {
-    if (window_size_suitable()) {
+    if (width > NSAMP) {
```

Comment:
This should be `plotwidth`:

* `width` is the total width of the terminal window
* `plotwidth`, which is set to `width - 4` by `paint_plot()`, is the width of the area used for plotting the data.

Within `plot_values()`, the data arrays are accessed through an index that wraps modulo `plotwidth`.

Reply:
Hm, you are correct - this part of the PR looks misguided.

I do remember triggering a segfault, but in hindsight it's possible I triggered the exit(0) path and mistook it for something worse without following up.

This feels like a distraction - unless you have a preference, I think I'll back the overrange check out of the negative-value PR unless I can more carefully trigger a bug.